src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/Marshal/GetFunctionPointerForDelegateTests.cs

```diff
@@ -92,8 +92,27 @@ public void GetFunctionPointer_GenericDelegate_ThrowsArgumentException()
             AssertExtensions.Throws<ArgumentException>("delegate", () => Marshal.GetFunctionPointerForDelegate(d));
         }
 
+        [Fact]
+        public void GetFunctionPointerForDelegate_MarshalledOpenStaticDelegate()
+        {
+            MethodInfo targetMethod = typeof(GetFunctionPointerForDelegateTests).GetMethod(nameof(Method), BindingFlags.NonPublic | BindingFlags.Static);
+            Delegate original = targetMethod.CreateDelegate(typeof(NonGenericDelegate), null);
+            IntPtr ptr = Marshal.GetFunctionPointerForDelegate(original);
+            Assert.NotEqual(IntPtr.Zero, ptr);
+        }
+
+        [Fact]
+        public void GetFunctionPointerForDelegate_MarshalledClosedStaticDelegate()
+        {
+            MethodInfo targetMethod = typeof(GetFunctionPointerForDelegateTests).GetMethod(nameof(Method), BindingFlags.NonPublic | BindingFlags.Static);
+            Delegate original = targetMethod.CreateDelegate(typeof(NoArgsDelegate), "value");
+            IntPtr ptr = Marshal.GetFunctionPointerForDelegate(original);
+            Assert.NotEqual(IntPtr.Zero, ptr);
+        }
+
         public delegate void GenericDelegate<T>(T t);
         public delegate void NonGenericDelegate(string t);
+        public delegate void NoArgsDelegate();
 
         private static void Method(string s) { }
```

Comment:
@lateralusX 
So I got a crash in a sample (in an unpatched runtime) if I change this test slightly:

```csharp
private static void Method2(string s, [MarshalAs(UnmanagedType.LPStr)] string s2) { }

public delegate void StringArgDelegate(string s2);

...
            MethodInfo targetMethod = typeof(GetFunctionPointerForDelegateTests).GetMethod(nameof(Method2), BindingFlags.NonPublic | BindingFlags.Static);
            Delegate original = targetMethod.CreateDelegate(typeof(StringArgDelegate), "value");
            IntPtr ptr = Marshal.GetFunctionPointerForDelegate(original);
            Assert.NotEqual(IntPtr.Zero, ptr);
```

Reply:
I guess this is just incidental due to the behavior of `malloc` on various systems, since the underlying issue is a read past the end of the allocated region.

Reply:
Jupp, if you run with a debug allocator, putting 0xFF in 4 bytes past allocation, you should get original test to crash before this PR. If you run on release or with an allocator not doing that, the behavior will be random.